subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -360,6 +360,38 @@ static struct bass_recv_state_internal *get_free_recv_state(void)
 		}
 	}
 
+	LOG_DBG("All receive states occupied");
+
+#if defined(CONFIG_BT_BAP_SCAN_DELEGATOR_OVERWRITE_RECV_STATE)
+	/* Try to free an inactive receive state*/
+	for (size_t i = 0U; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
+		struct bass_recv_state_internal *free_internal_state =
+					&scan_delegator.recv_states[i];
+
+		struct bt_bap_scan_delegator_recv_state *state = &free_internal_state->state;
+		/* Check the bis_sync for each subgroup. If it is 0 for all subgroups,
+		 * the receive state is currently not in sync with any stream.
+		 */
+		bool bis_sync_set = false;
+		for (size_t j = 0; j < state->num_subgroups; j++) {
+			struct bt_bap_scan_delegator_subgroup *subgroup = &state->subgroups[j];
+			if (subgroup->bis_sync != 0U ) {
+				bis_sync_set = true;
+				break;
+			}
+		}
+
+		if (bis_sync_set == false) {
+			LOG_DBG("Receive state at index %d is currently inactive. "
+					"This state will be freed and used", i);
+			memset(free_internal_state, 0, sizeof(*free_internal_state));
+			return free_internal_state;
+		}
```

Comment:
Might be better to add a reference to the broadcast sink that is associated with the receive state (if any), and then check `sink-> big == NULL`, as does not taking the case where it attempting to sync to the BIG into account